### DISCUSSION TITLE
DVR

### DISCUSSION THREAD

**Discussion by klasadoroty** [ant-media/Ant-Media-Server#5875] [General]
Hi, does the described configuration also work in the Community version? :
https://antmedia.io/how-to-enable-dvr-in-hls/
We did a test, configured as described and ... nothing happened.

**Comment by USAMAWIZARD** [comment]
Hi @klasadoroty , 
Yes it should work in community edition as well , Which AMS version are you using right now.

**Reply by klasadoroty** [reply]
2.7.0 , community edition

**Reply by USAMAWIZARD** [reply]
let me test it and get back to you.

**Reply by klasadoroty** [reply]
Hi,
..I understand that these are very advanced tests.. I guess I didn't want to involve you so much.
Regards

**Reply by USAMAWIZARD** [reply]
@klasadoroty  sorry It slipped my mind , I have tried it on my end and its working for me , can you please  let me know exactly what error do you get or which specific step is not working for you.

**Comment by klasadoroty** [comment]
I repeated the configuration:
settings.hlsPlayListType=event
settings.hlsflags=
I restarted AMS.
/by the way:
searching for the string: 'settings.hlsflags=' is pointless because it is: 'hlsflags=delete_segments'/
now the 'ts' files appear... but how to recreate it so I can go back?
In the player I can only rewind a few seconds.
Any player configuration?

**Reply by klasadoroty** [reply]
I think it's already working...
as I wrote:
searching for the string: 'settings.hlsflags=' is pointless because it is: 'hlsflags=delete_segments'
is also:
'hlsPlayListType='
and it says that you need to search for a string of characters:
settings.hlsPlayListType=event
settings.hlsflags=

and... a little confusion...
It seems to be working fine now.

**Reply by yashtandon113** [reply]
@klasadoroty 
It is good to know that it is working fine for you now.

What I understand is that you are changing the settings in the application's advanced settings. This is a new feature implemented since v2.6.2 of AMS. Check out the following document for configuring the application settings.
https://antmedia.io/docs/guides/configuration-and-testing/ams-application-configuration/

Before v2.6.2, some advanced settings could only be changed from the backend, so you need a prefix like **settings.hlsflags** and similar for other settings.

I hope it is clear for you.

**Reply by klasadoroty** [reply]
Here's another question:
how much the DVR affects CPU or memory/usage ?

**Reply by USAMAWIZARD** [reply]
It Depends on Number of Cameras and number of viewers at a time , Can you please let me know how many camera and concurrent
 playing will be there then I can give you an estimate of the CPU requirement

**Reply by klasadoroty** [reply]
Number of Cameras - 1
number of viewers at a time - max 1000
or..
Number of Cameras - 10
number of viewers at a time - max 100

**Reply by yashtandon113** [reply]
@klasadoroty 
For 1000 simultaneous HLS viewers, one 4-core dedicated CPU-optimized server with 8 GB would be enough, in my opinion.

The important thing for 1000 HLS viewers is that bandwidth should be enough, and for storing all the files for the DVR, make sure that disk space is also enough.

**Reply by klasadoroty** [reply]
HI,
we do tests and...
after transmitting, .ts .. files are not deleted.
With every transmission, the number of files increases.
They should -probably- be deleted.
Both for options 1 and 2
When should files be deleted after completion?
For example, does a camera failure and no stream result in files being deleted?

..something is happening... -I stopped the stream from the camera ..a few minutes and restarted it - the files were deleted. 
!! note - the stream has been stopped
Please write precisely the operating/deleting rules etc.
1 - stream operating 24/7 - rule
2 - stream / and DVR stream running for e.g. 5 hours

**Reply by USAMAWIZARD** [reply]
if this is set to true   `settings.deleteHLSFilesOnEnded=true`
Files will will be deleted once the streaming has been stop from the camera or the connection get lost.
if you want to keep the files even after the stream has been stop you can set ` settings.deleteHLSFilesOnEnded=false`
